### PR TITLE
optimize: add context to huawei/obs

### PR DESCRIPTION
**Pull request by 1046102779**
Signed-off-by: 1046102779 <seachen@tencent.com>

### CODE REVIEW

File: bindings/huawei/obs/obs_test.go
Comment (ItalyPaleAle) [inline bindings/huawei/obs/obs_test.go:268]: Please use `context.Background()` in tests

File: bindings/huawei/obs/obs_service.go
Comment (ItalyPaleAle) [inline bindings/huawei/obs/obs_service.go:39]: You need to add a second argument `obs.WithRequestContext(ctx)`. See https://github.com/ItalyPaleAle/dapr-components-contrib/blob/bindings-context/bindings/huawei/obs/obs_service.go#L39

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Thanks for this! Funnily enough, I was literally working on this too (as part of [another PR](https://github.com/ItalyPaleAle/dapr-components-contrib/tree/bindings-context/bindings/huawei/obs)). But your contribution is still good and appreciated.

Just a couple of small things to make sure contexts are actually, fully used.

File: bindings/huawei/obs/obs_test.go
Comment (1046102779) [reply bindings/huawei/obs/obs_test.go:268]: image
How should we unify?  `TODO()` or `Background()`

File: bindings/huawei/obs/obs_service.go
Comment (1046102779) [reply bindings/huawei/obs/obs_service.go]: Hah, thanks.

File: bindings/huawei/obs/obs_test.go
Comment (ItalyPaleAle) [reply bindings/huawei/obs/obs_test.go]: Background is better. TODO is equivalent but it's meant as "placeholder". If you know you want a background context, use Bakcground.

Don't worry about the other files.... we'll get to that one day :)

**Comment by ItalyPaleAle** [comment]
@berndverst  please do not squash this PR when you merge it :)

**Comment by daixiang0** [comment]
@artursouza @ItalyPaleAle  seems `automerge` bot does not work as expected.

**Comment by ItalyPaleAle** [comment]
@daixiang0 I added "autoupdate" but not "automerge". As an approver (and not maintainer) I am not allowed to add "automerge". If you don't have an issue with this PR perhaps you or @berndverst  can merge it?

**Comment by berndverst** [comment]
@ItalyPaleAle he cannot merge it. @daixiang0 is not a maintainer of contrib, only an approver!

**Comment by berndverst** [comment]
@daixiang0 @ItalyPaleAle only maintainers are allowed to apply the `automerge` label. If anyone else adds that label this will have no affect. That is by design.

**Review by berndverst — APPROVED** [review]

**Pull request merged**
